src/domain/helpers/ExcelBuilder.ts

```diff
@@ -335,7 +335,8 @@ export class ExcelBuilder {
 
             const newTEIs = _.difference(allTEIs, existingTEIs);
 
-            for (const id of newTEIs) {
+            return await promiseMap(newTEIs, async (id, index) => {
+                const rowStart = dataSource.dataValues.rowStart + newTEIs.length - index - 1; // reverse tei ids
```

Comment:
`rowStart` was defined at the beginning of the code and again inside the promiseMap which could confusing when you're reading the code since you don't know exactly what variable we're referring to.
```ts
let { rowStart } = dataSource.dataValues;
```

just to make sure what variable we're using I think it would be easy to reuse this variable and rename inside the promiseMap to  `teiRowStart`. 

```ts
const teiRowStart = rowStart + newTEIs.length - index - 1; // reverse tei ids
const cells = await this.excelRepository.getCellsInRange(template.id, {
  ...dataSource.dataValues,
  rowStart: teiRowStart ,
  rowEnd: teiRowStart,
});
```

What do you think?

Reply:
Good suggestion. it'll definitely make the code easier to read 👌